lib/releases.sh

```diff
@@ -1,44 +1,26 @@
 #!/bin/bash
 
 function get_latest_release() {
-  # get last page
+  # set url to get all releases from github
+  url="${1}?per_page=1000"
+  
+  # This gets the latest release as vx.y.z or vx.y.z-rc.0, including any version with a suffix starting with - , for example -rc.0
+  # The order is exactly as released in Github.
+  # Downside is that selecting official releases only isn't possible, while pre-release
+  # selection is possible given specific enough prefix, like v1.3.0-pre
+
   if [ -z "${GITHUB_TOKEN:-}" ]; then
-    last_page="$(curl -s -I "${1}" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g')"
+    release_tag=$(curl -sL "${url}" | jq .[].name -r | egrep "^v1.4." -m 1)
   else
-    last_page="$(curl -s -I "${1}" -H "Authorization: token $GITHUB_TOKEN" | grep '^link:' | sed -e 's/^link:.*page=//g' -e 's/>.*$//g')"
+    release_tag=$(curl -sL "${url}" -H "Authorization: token $GITHUB_TOKEN"  | jq .[].name -r | egrep "^v1.4." -m 1)
   fi
-  # default last page to 1
-  last_page="${last_page:-1}"
-
-  set +x
-
-  for current_page in $(seq 1 "$last_page"); do
-
-    url="${1}?page=${current_page}"
-
-    if [ -z "${GITHUB_TOKEN:-}" ]; then
-      release="$(curl -sL "${url}")" || { set -x && exit 1; }
-    else
-      release="$(curl -H "Authorization: token ${GITHUB_TOKEN}" -sL "${url}")" || { set -x && exit 1; }
-    fi
-
-    # This gets the latest release as vx.y.z or vx.y.z-rc.0, including any version with a suffix starting with - , for example -rc.0
-    # The order is exactly as released in Github.
-    # Downside is that selecting official releases only isn't possible, while pre-release
-    # selection is possible given specific enough prefix, like v1.3.0-pre
-    release_tag="$(echo "${release}" | jq -r "[.[].tag_name | select( startswith(\"${2:-}\"))] | .[0]")"
-    # if release tag found
-    if [[ "${release_tag}" != "null" ]]; then
-      break
-    fi
-
-  done
 
   set -x
   # if not found
   if [[ "${release_tag}" == "null" ]]; then
```

Comment:
This null check probably doesn't work anymore, as `release_tag` is no longer coming from `select()` of jq which produced the `null`.

Reply:
removed